src/core/connection.c

```diff
@@ -2488,6 +2687,14 @@ QuicConnProcessPeerTransportParameters(
     }
 
     if (!FromResumptionTicket) {
+        if (Connection->Settings.VersionNegotiationExtEnabled &&
+            Connection->PeerTransportParams.Flags & QUIC_TP_FLAG_VERSION_NEGOTIATION) {
+            QUIC_STATUS Status = QuicConnProcessPeerVersionNegotiationTP(Connection);
+            if (QUIC_FAILED(Status)) {
```

Comment:
We have an error case label here. Is this not expected to be an error case, but actually a special early return case, I'd like a comment here and potentially at the end of the function incase we ever do an allocation or something in here.

Reply:
you think it would be better to add a `goto Error` and put an `Error` label at the bottom?

Reply:
Theres already an error label at the bottom of this function. This early return is going to bypass that.